bm_rbr_data_msg.cpp

```diff
@@ -0,0 +1,233 @@
+#include "bm_rbr_data_msg.h"
+#include <math.h>
+
+namespace BmRbrDataMsg {
+
+CborError encode(Data &d, uint8_t *cbor_buffer, size_t size,
+                 size_t *encoded_len) {
+  CborError err;
+  CborEncoder encoder, map_encoder;
+  cbor_encoder_init(&encoder, cbor_buffer, size, 0);
+
+  do {
+    // Set the unused field to NaN
+    switch (d.sensor_type) {
+    case TEMPERATURE: {
+      d.pressure_deci_bar = NAN;
+      break;
+    }
+    case PRESSURE: {
+      d.temperature_deg_c = NAN;
+      break;
+    }
+    case PRESSURE_AND_TEMPERATURE: {
+      break;
+    }
+    default: {
+      d.temperature_deg_c = NAN;
+      d.pressure_deci_bar = NAN;
```

Comment:
nit just incase someone adds in more cases after default in the future.
```suggestion
      d.pressure_deci_bar = NAN;
      break;
```

Reply:
Oof what anarchy 😅